src/evo/deterministicmns.cpp

```diff
@@ -645,6 +646,15 @@ bool CDeterministicMNManager::ProcessBlock(const CBlock& block, gsl::not_null<co
         updatesRet = {newList, oldList, diff};
     }
 
+    ::g_stats_client->gauge("masternodes.count", newList.GetAllMNsCount());
+    ::g_stats_client->gauge("masternodes.weighted_count", newList.GetValidWeightedMNsCount());
+    ::g_stats_client->gauge("masternodes.enabled", newList.GetValidMNsCount());
+    ::g_stats_client->gauge("masternodes.weighted_enabled", newList.GetValidWeightedMNsCount());
+    ::g_stats_client->gauge("masternodes.evo.count", newList.GetAllEvoCount());
+    ::g_stats_client->gauge("masternodes.evo.enabled", newList.GetValidEvoCount());
+    ::g_stats_client->gauge("masternodes.mn.count", newList.GetAllMNsCount() - newList.GetAllEvoCount());
+    ::g_stats_client->gauge("masternodes.mn.enabled", newList.GetValidMNsCount() - newList.GetValidEvoCount());
```

Comment:
Many of these `Get*Count()`s are essentially loops, probably makes sense to wrap them in `if (::g_stats_client->active()) {...}` to avoid extra work here for nodes running with stats disabled (`ProcessBlock()`'s performance is critical for IBD)